pkg/kubelet/server/stats/handler.go

```diff
@@ -37,8 +37,27 @@ import (
 	"k8s.io/kubernetes/pkg/volume"
 )
 
+// MetadataPovider provid basic metadata needed by stats handlers.
```

Comment:
I'm curious if it would be possible to give this interface a more descriptive name than `MetadataProvider`? Metadata is a very generic term, so for someone reading this code, it could be difficult to know this interface's purpose.

Reply:
Also, can you share more about the rationale behind exporting `MetadataProvider`? I believe its only used in the `stats` package?

Reply:
> I'm curious if it would be possible to give this interface a more descriptive name than `MetadataProvider`? Metadata is a very generic term, so for someone reading this code, it could be difficult to know this interface's purpose

As `MetadataProvider` type is located in `stats` package, so the name will be used as `stats.MetadataPovider` by other packages. It's clear enough: the metadata provider of stats. If you find other fancy name, I would like to change this.

> Also, can you share more about the rationale behind exporting `MetadataProvider`? I believe its only used in the `stats` package?

It also used by `pkg/kubelet/server/stats` package in this change.